#!/usr/bin/env python
# -*- coding:UTF-8 -*-
PACKAGE = "auto_drive"#当前包名字

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()#参数生成器

gen.add("HeadKp", double_t, 0, "Heading Proportional Gain", 0.0, -100.0, 100.0)
gen.add("HeadKi", double_t, 0, "Heading Integral Gain", 0.0, -100.0, 100.0)
gen.add("HeadKd", double_t, 0, "Heading Derivative Gain", 0.0, -100.0, 100.0)
gen.add("VelKp", double_t, 0, "Velicity Proportional Gain", 0.0, -100.0, 100.0)
gen.add("VelKi", double_t, 0, "Velicity Integral Gain", 0.0, -100.0, 100.0)
gen.add("VelKd", double_t, 0, "Velicity Derivative Gain", 0.0, -100.0, 100.0)
gen.add("targetvel", double_t, 0, "Velicity Integral Gain", 0.0, -100.0, 100.0)
gen.add("targethead", double_t, 0, "Velicity Derivative Gain", 0.0, -100.0, 100.0)
gen.add("MODE", int_t, 0, "Mode", 0, 0, 10)

exit(gen.generate(PACKAGE, "auto_drive", "PID"))
